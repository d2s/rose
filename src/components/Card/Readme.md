### Card example

#### Default card (without props)

```js
<Card />
```


#### Card example with a custom title (without defined content)

```js
<Card title="Current Rate" />
```

#### Card with a Chart

```js
<Card title="7d History">
  <Chart />
</Card>
```
