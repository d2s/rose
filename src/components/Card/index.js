import React, { Component } from 'react';
import PropTypes from 'prop-types';
import glamorous, {ThemeProvider} from 'glamorous';

// Load required components from other directories
// import ComparisonSelector from '../ComparisonSelector';


// CardWrapper
const CardWrapper = glamorous.div({
  backgroundColor: '#fff',
  color: '#000',
  padding: 0,
  textAlign: 'left',
  margin: 10,
  borderRadius: 5,
  borderColor: '#000',
  borderWidth: 1,
  borderStyle: 'solid'
});

// Header wrapper
const CardHeader = glamorous.header({
  backgroundColor: 'gray',
  color: '#000',
  padding: 0,
  display: 'block',
  width: 'auto',
  textAlign: 'left'
});

// h1 Main title
const CardTitle = glamorous.h1({
  backgroundColor: '#222',
  color: '#fff',
  width: 'auto',
  padding: 15,
  margin: 0,
  display: 'inline-block',
  fontSize: '1.2em',
  fontWeight: 500
});

// Card content wrapper
const CardContent = glamorous.div({
  backgroundColor: '#fff',
  color: '#000',
  padding: 10,
  display: 'block',
  width: 'auto',
  textAlign: 'left',
  borderBottomLeftRadius: 5,
  borderBottomRightRadius: 5
});

/**
 * Card component for displaying application's data content.
 */
class Card extends Component {
  static propTypes = {
    /** Text string for the card title, with a fallback defaultProps text if not specified. */
    title: PropTypes.string,
    /** Card content, with a fallback defaultProps if not specified. */
    content: PropTypes.any
  }
  static defaultProps = {
    title: 'Card title',
    content: null
  }

  render() {
    return (
      <CardWrapper>
        <CardHeader>
          <CardTitle>
            {this.props.title}
          </CardTitle>
        </CardHeader>
        <CardContent>
          {this.props.children}
        </CardContent>
      </CardWrapper>
    );
  }
}

export default Card;
