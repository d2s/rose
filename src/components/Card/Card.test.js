import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'

import Card from './index';

describe('<Card />', () => {
  describe('with ReactDOM', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Card />, div);
      ReactDOM.unmountComponentAtNode(div);
    });
  });

  describe('with Enzyme', () => {
    it('shallow renders without crashing', () => {
      shallow(<Card />);
    });
  });

  describe('with Snapshot Testing', () => {
    it('renders the same way', () => {
      const component = renderer.create(<Card />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    });
  });
});
