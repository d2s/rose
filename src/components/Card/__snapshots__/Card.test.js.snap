// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> with Snapshot Testing renders the same way 1`] = `
<div
  className="css-1oq7lj6"
>
  <header
    className="css-7i6bp6"
  >
    <h1
      className="css-1x5veox"
    >
      Card title
    </h1>
  </header>
  <div
    className="css-1ryc7tm"
  />
</div>
`;
